spec/controllers/contributions_controller_spec.rb

```diff
@@ -45,9 +47,103 @@
         expect(page.find('table')).to have_content(contribution_list[0].link)
                                   .and have_content(contribution_list[0].pr_state_text)
                                   .and have_content(contribution_list[0].created_at.strftime('%d/%m/%Y'))
+                                  .and have_selector(:css, "a[href='/contributions/#{contribution_list[0].id}/edit']")
                                   .and have_content(contribution_list[1].link)
                                   .and have_content(contribution_list[1].pr_state_text)
                                   .and have_content(contribution_list[1].created_at.strftime('%d/%m/%Y'))
+                                  .and have_selector(:css, "a[href='/contributions/#{contribution_list[1].id}/edit']")
+      end
+    end
+  end
+
+  describe 'GET edit' do
+    let(:contribution) { create(:contribution) }
+    let(:params) { { id: contribution.id } }
+
+    before do
+      allow(user.contributions).to receive(:find).with(contribution.id.to_s) { contribution }
+    end
+
+    context 'when the page is accessed' do
+      it 'is expected to return the successful status' do
+        get :edit, params: params
+
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'is expected to render the edit page' do
+        get :edit, params: params
+
+        expect(page.find(:css, '.main'))
+          .to have_content('Editando Contribuição')
+          .and have_content(contribution.link)
+          .and have_field('contribution[description]')
+          .and have_selector("input[type='submit']")
+      end
+    end
+  end
+
+  describe 'PUT update' do
+    let(:contribution) { create(:contribution) }
+    let(:params) do
+      {
+        id: contribution.id,
+        contribution: {
+          description:
```

Comment:
add the `description: description` just to be consistent with the other parts of the code